src/Plugin/media/Source/MediaAvPortalVideo.php

```diff
@@ -0,0 +1,254 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\media_avportal\Plugin\media\Source;
+
+use Drupal\Component\Utility\Crypt;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Entity\Display\EntityFormDisplayInterface;
+use Drupal\Core\Entity\EntityFieldManagerInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Field\FieldTypePluginManagerInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Logger\LoggerChannelInterface;
+use Drupal\Core\Messenger\MessengerInterface;
+use Drupal\media\MediaSourceBase;
+use Drupal\media\MediaInterface;
+use Drupal\media\MediaTypeInterface;
+use Drupal\media_avportal\AvPortalClient;
+use Drupal\media_avportal\AvPortalResource;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides a media source plugin for Media AV Portal resources.
+ *
```

Comment:
Let's add a comment to state that most of this code comes from `\Drupal\media\Plugin\media\Source\OEmbed`.

Reply:
No, please. Not needed.

Reply:
😢